test/GPv2UniswapRouter.test.ts

```diff
@@ -76,11 +107,259 @@ describe("GPv2UniswapRouter", () => {
+          partiallyFillable: false,
+          signingScheme: SigningScheme.EIP712,
+        }),
+        executedAmount: 0,
+        signature: `0x${"51".repeat(65)}`,
+        ...flags,
+      };
+    }
+
+    it("reverts for paths without at least two tokens", async () => {
+      await expect(
+        uniswapRouter.settleSwap(
+          [fillAddress(1)],
+          sampleTrade({
+            kind: OrderKind.SELL,
+            sellTokenIndex: 0,
+            buyTokenIndex: 0,
+            sellAmount: ethers.constants.Zero,
+            buyAmount: ethers.constants.Zero,
+          }),
+        ),
+      ).to.be.revertedWith("invalid path");
+    });
+
+    it("reverts for trades that do not correspond to specified path", async () => {
+      await expect(
+        uniswapRouter.settleSwap(
+          [fillAddress(1), fillAddress(2)],
+          sampleTrade({
+            kind: OrderKind.SELL,
+            // NOTE: The trade is going in the opposite direction of the path.
+            sellTokenIndex: 1,
+            buyTokenIndex: 0,
+            sellAmount: ethers.constants.Zero,
+            buyAmount: ethers.constants.Zero,
+          }),
+        ),
+      ).to.be.revertedWith("invalid trade for path");
+    });
+
+    it("executes a GPv2 settlement", async () => {
+      const pair = await deployMockPair(fillAddress(1), fillAddress(2));
+      const [reserveSell, reserveBuy] = [
+        ethers.utils.parseEther("200000.0"),
+        ethers.utils.parseEther("100.0"),
+      ];
+
+      const sellAmount = ethers.utils.parseEther("1.0");
+      const amountOut = getAmountOut(sellAmount, reserveSell, reserveBuy);
+
+      const trade = sampleTrade({
+        kind: OrderKind.SELL,
+        sellTokenIndex: 0,
+        buyTokenIndex: 1,
+        sellAmount,
+        buyAmount: ethers.utils.parseEther("0.0003"),
+      });
+
+      await pair.mock.getReserves.returns(reserveSell, reserveBuy, 0);
+      await settlement.mock.settle
+        .withArgs(
+          [fillAddress(1), fillAddress(2)],
+          [amountOut, sellAmount],
+          [trade],
+          [
+            [],
+            [
+              {
+                target: fillAddress(1),
+                value: ethers.constants.Zero,
+                callData: IERC20.encodeFunctionData("transfer", [
+                  pair.address,
+                  sellAmount,
+                ]),
+              },
+              {
+                target: pair.address,
+                value: ethers.constants.Zero,
+                callData: IUniswapV2Pair.encodeFunctionData("swap", [
+                  0,
+                  amountOut,
+                  settlement.address,
+                  "0x",
+                ]),
+              },
+            ],
+            [],
+          ],
+        )
+        .returns();
+
+      await expect(
+        uniswapRouter.settleSwap([fillAddress(1), fillAddress(2)], trade),
+      ).to.not.be.reverted;
+    });
+
+    describe("Swap Amounts", () => {
```

Comment:
What's the difference between this and the previous test? The length of the path? Maybe you can update the description?

Reply:
This tests the amounts are correctly computed for both buy and sell tokens. The previous test is not so useful.

Reply:
Fixed in #558 (sorry messed up when rebasing :see_no_evil:)